app-libs/parentchain-interface/src/integritee/api_client_types.rs

```diff
@@ -0,0 +1,95 @@
+	limitations under the License.
+
+*/
+
+//! Contains semi-generic type definitions to talk to the node without depending on an implementation of Runtime.
+//!
+//! You need to update this if you have a signed extension in your node that
+//! is different from the integritee-node, e.g., if you use the `pallet_asset_tx_payment`.
+
+pub use itp_types::parentchain::{
+	AccountData, AccountId, AccountInfo, Address, Balance, Hash, Index, Signature as PairSignature,
+};
+use sp_runtime::generic;
+use substrate_api_client::ac_primitives::{PlainTipExtrinsicParams, WithExtrinsicParams};
+pub use substrate_api_client::{
+	ac_node_api::{
+		metadata::{InvalidMetadataError, Metadata, MetadataError},
+		EventDetails, Events, StaticEvent,
+	},
+	ac_primitives::{
+		config::{AssetRuntimeConfig, Config},
+		extrinsics::{
+			AssetTip, CallIndex, ExtrinsicParams, GenericAdditionalParams, GenericAdditionalSigned,
+			GenericExtrinsicParams, GenericSignedExtra, PlainTip, UncheckedExtrinsicV4,
+		},
+		serde_impls::StorageKey,
+		signer::{SignExtrinsic, StaticExtrinsicSigner},
+	},
+	rpc::Request,
+	storage_key, Api,
+};
+
+// traits from the api-client
+pub mod traits {
+	pub use substrate_api_client::{GetAccountInformation, GetChainInfo, GetStorage};
+}
+
+pub type IntegriteeTip = PlainTip<Balance>;
+pub type IntegriteeRuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, PlainTipExtrinsicParams<AssetRuntimeConfig>>;
+
+// Configuration for the ExtrinsicParams.
+//
+// Pay in asset fees.
+//
+// This needs to be used if the node uses the `pallet_asset_tx_payment`.
+pub type IntegriteeExtrinsicParams = GenericExtrinsicParams<IntegriteeRuntimeConfig, IntegriteeTip>;
+pub type IntegriteeAdditionalParams = GenericAdditionalParams<IntegriteeRuntimeConfig, Hash>;
+
+pub type IntegriteeSignedExtra = GenericSignedExtra<IntegriteeTip, Index>;
+pub type IntegriteeSignature = Signature<IntegriteeSignedExtra>;
+
+pub type IntegriteeUncheckedExtrinsic<Call> =
+	UncheckedExtrinsicV4<Address, Call, PairSignature, IntegriteeSignedExtra>;
+
+/// Signature type of the [UncheckedExtrinsicV4].
+pub type Signature<SignedExtra> = Option<(Address, PairSignature, SignedExtra)>;
+
+pub type Block = generic::Block<Header, IntegriteeUncheckedExtrinsic<([u8; 2])>>;
+
+#[cfg(feature = "std")]
+pub use api::*;
+use itp_types::parentchain::Header;
+
+#[cfg(feature = "std")]
+mod api {
+	use super::IntegriteeRuntimeConfig;
+	use crate::ParentchainApiTrait;
+	use itp_node_api::api_client::AccountApi;
+	use itp_types::parentchain::{AccountId, Balance, Hash, Index};
+	pub use substrate_api_client::{
+		api::Error as ApiClientError,
+		rpc::{tungstenite_client::TungsteniteRpcClient, Error as RpcClientError},
+	};
+	use substrate_api_client::{
+		Api, GetBalance, GetChainInfo, GetStorage, GetTransactionPayment, SubscribeEvents,
+	};
+
+	pub type IntegriteeApi = Api<IntegriteeRuntimeConfig, TungsteniteRpcClient>;
+
+	impl ParentchainApiTrait for IntegriteeApi {}
```

Comment:
Super traits are usually generically implemented for `T` satisfying all the constraints, i.e.,

```rust
impl<T> ParentchainApiTrait for T 
where T: ....
{}
```